client/app/components/packages/pas-form/edit.hbs

```diff
@@ -697,22 +697,29 @@
           <button
             type="button"
             class="button expanded large"
-            disabled={{not this.isSaveable}}
-            {{on "click" this.save}}
+            disabled={{or (not this.isSaveable) this.isSavingOrSubmitting}}
```

Comment:
I think the truth helper here makes this a little confusing and spread out because we already have a lot of conditional logic for determining the saveable/submittable button state  inside those getters.

Can we build this logic into `isSaveable`? Or another getter?

Reply:
I think we need to keep isSaveable separate because it represents another state altogether -- whether someone can click Save.

What do you mean by another getter?

Reply:
Oh I see what you mean by building it into isSaveable/isSubmittable.

I think it's nice to have  `this.isSavingOrSubmitting` (or something that makes more sense to everyone) 
because later we can use this to succinctly disable inputs. For example
```
<Input 
 disabled={{this.isSavingOrSubmitting}}
```

I was thinking isSavingORSubmitting to be explicit about what is happening. instead of an opaque variable name like `isProcessing`.

How about this? 
```
  get isSaveable() {
    const {
      isDirty: isPasFormDirty,
      isValid: isPasFormValid,
    } = this.saveableChanges;
    const {
      isBblsDirty,
      isApplicantsDirty,
    } = this.pasForm;
     
    const isNotPatching = !(this.save.isRunning || this.submit.isRunning)
    return ((isPasFormDirty && isPasFormValid) || isBblsDirty || isApplicantsDirty) && isNotPatching;
  }
```

Reply:
But how? It's telling us whether or not the form is saveable, and we want to prevent saveability if it's already in the process of saving.

Another getter like 
```
get theSaveButtonIsNotDisabled() {
  return !this.isSaveable || this.isSavingOrSubmitting;
}
```

I think it works just as well to keep that conditional logic in the saveable getter, though.

Reply:
> I was thinking to name it isSavingOrSubmitting to be explicit about what is happening. Vs. of an opaque variable name like isProcessing.

Good point, agree.


I like your code sample `const isNotPatching = !(this.save.isRunning || this.submit.isRunning)` but I think it's easier to follow if things are written so as to avoid negatives/double negatives:

Could we make this work with `isPatching` instead? :) :) :)

Reply:
I think I understood and gave it a go! Thanks 💯 for this suggestion. I think it looks much cleaner after a refactor